corehq/form_processor/tests/test_kafka.py

```diff
@@ -73,3 +76,41 @@ def test_duplicate_case_published(self):
         case_meta = change_meta_from_kafka_message(case_consumer.next().value)
         self.assertEqual(case_id, case_meta.document_id)
         self.assertEqual(self.domain, case_meta.domain)
+
+    def test_duplicate_ledger_published(self):
+        # setup products and case
+        product_a = make_product(self.domain, 'A Product', 'prodcode_a')
+        product_b = make_product(self.domain, 'B Product', 'prodcode_b')
+        case_id = uuid.uuid4().hex
+        form_xml = get_simple_form_xml(uuid.uuid4().hex, case_id)
+        submit_form_locally(form_xml, domain=self.domain)[1]
+
+        # submit ledger data
+        balances = (
+            (product_a._id, 100),
+            (product_b._id, 50),
+        )
+        ledger_blocks = [
+            get_single_balance_block(case_id, prod_id, balance)
+            for prod_id, balance in balances
+        ]
+        form = submit_case_blocks(ledger_blocks, self.domain)
+
+        # submit duplicate
+        ledger_consumer = get_test_kafka_consumer(topics.LEDGER)
+        dupe_form = submit_form_locally(form.get_xml(), domain=self.domain)[1]
+        self.assertTrue(dupe_form.is_duplicate)
+
+        # confirm republished
+        ledger_meta_a = change_meta_from_kafka_message(ledger_consumer.next().value)
+        ledger_meta_b = change_meta_from_kafka_message(ledger_consumer.next().value)
```

Comment:
the consumer's default is to start at the end of the feed so it will only have changes that are published after it is defined (that's why it's defined after the first submission in all of these tests)